File: Boj.단계별/14_백트래킹/python/15650.py
n,m = map(int,input().split())
def search(num,cnt,arr):
    if cnt == m:
        print(" ".join(arr))
        return
    for i in range(num+1,n+1):
        search(i,cnt+1,arr+[str(i)])

for i in range(1,n+1):
    search(i,1,[str(i)])

File: Boj.단계별/14_백트래킹/javascript/15650.js
const [n,m] = require('fs').readFileSync('/dev/stdin').toString().split('\n')[0].split(' ').map(Number);

let checkSet = new Set();
let checkCnt = new Array(n).fill(false);
const solution = (n,cnt,target,acc) => {
  if (cnt === target){ 
    newStr = acc.sort((a,b) => Number(a)-Number(b)).join("")
    if (checkSet.has(newStr)) return false;
    checkSet.add(newStr);
    return acc.join(" ") + "\n";
  }
  let result = "";
  for (let i = 1 ; i<=n ; i++){
    if (checkCnt[i]) continue
    checkCnt[i] = true;
    newStr = solution(n,cnt+1,target,acc.concat(i));
    if (newStr) result += newStr;
    checkCnt[i] = false;
  }
  return result;
};
console.log(solution(n,0,m,[]));
